docs/multitenancy.md

```diff
@@ -0,0 +1,81 @@
+# Multi Tenancy
+
+Knative Eventing generally provides two sets of data-plane components:
+
+- single-tenant: these are components capable of handling one and only one
+  resource. For instance both the `APIServerSource` and the `GitHubSource`
+  adapters are single-tenant. There is one component per resource and usually
+  the component is deployed in the same namespace as the resource it manages.
+- multi-tenant: these are components capable of handling more than one resource.
+  For example, both the in-memory and the kafka dispatchers support
+  multi-tenancy. These components can typically be either deployed in the system
+  namespace (eg. `knative-eventing`) or in the same namespace as the resources
+  it manages.
+
+## Configuring multi-tenancy
+
+Each component potentially supports three multi-tenancy modes:
+
+- single-tenant
+- multi-tenant, one per namespace
+- multi-tenant, one per cluster
+
+By default all components operate in multi-tenant, one per cluster mode, unless
+the component does not support it. See the table below for more details.
+
+The default mode can be overridden by adding the `eventing.knative.dev/scope`
```

Comment:
I think we need more discussion on how to expose this to the user. Ville's recent multi-tenant broker implementation seems to suggest using `brokerclass`. I hope we can have a consistent UX.

Reply:
I agree!

Reply:
As was discussed in the original issue, classes was deemed to be the preferred approach.